sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -254,19 +333,11 @@ export class KeyClient {
    * @param {DeleteKeyOptions} [options] Optional parameters for the underlying HTTP request.
    */
   private async deleteKey(name: string, options: DeleteKeyOptions = {}): Promise<DeletedKey> {
-    const requestOptions = operationOptionsToRequestOptionsBase(options);
-    const span = this.createSpan("deleteKey", requestOptions);
+    let reqOptions: RequestOptionsBase = toRequestOptionsBase(options);
 
-    let response: DeleteKeyResponse;
-    try {
-      response = await this.client.deleteKey(
-        this.vaultUrl,
-        name,
-        this.setParentSpan(span, requestOptions)
-      );
-    } finally {
-      span.end();
-    }
+    const response = await trace<DeleteKeyResponse>("deleteKey", reqOptions, async (options) =>
+      this.client.deleteKey(this.vaultUrl, name, options)
+    );
```

Comment:
Even though I do like how many fewer lines this represents, I wonder what other design I could do instead. Something in my head is telling me "decorators!!", but I also could do some partial application like `trace(this.client.deleteKey)(this.vault, name, options)`. I can sense many combinations possible. I wonder what form of the possibilities could be the nicest.